diff --git a/shellinabox/shell_in_a_box.js b/shellinabox/shell_in_a_box.js
index 04aa100..2ca894d 100644
--- a/shellinabox/shell_in_a_box.js
+++ b/shellinabox/shell_in_a_box.js
@@ -164,6 +164,7 @@ ShellInABox.prototype.sendRequest = function(request) {
     request                  = new XMLHttpRequest();
   }
   request.open('POST', this.url + '?', true);
+  request.timeout = 30000; // Don't leave POST pending forever: force 30s timeout to prevent HTTP Proxy thread hijack
   request.setRequestHeader('Cache-Control', 'no-cache');
   request.setRequestHeader('Content-Type',
                            'application/x-www-form-urlencoded; charset=utf-8');
@@ -203,8 +204,10 @@ ShellInABox.prototype.onReadyStateChange = function(request) {
         this.sendRequest(request);
       }
     } else if (request.status == 0) {
-      // Time Out
-      this.sendRequest(request);
+      // Time Out or other connection problems: retry after 1s to prevent release CPU before retry
+      setTimeout(function(shellInABox) {
+                   return function() { shellInABox.sendRequest(); };
+                 } (this), 1000);
     } else {
       this.sessionClosed();
     }
--
2.1.0

